qa/vagrant/src/main/java/org/elasticsearch/packaging/test/ArchiveTestCase.java

```diff
@@ -61,4 +90,220 @@ public void test10Install() {
+            sh.bash("chmod +x '" + javaPath + "'");
+
+            assertThat(runResult.exitCode, is(1));
+            assertThat(runResult.stdout, containsString("could not find java; set JAVA_HOME or ensure java is in PATH"));
+        }
+    }
+
+    @Test
+    public void test40CreateKeystoreManually() {
+        assumeThat(installation, is(notNullValue()));
+
+        final Shell sh = new Shell();
+        if (Platforms.WINDOWS) {
+            // this is a hack around the fact that we can't run a command in the same session as the same user but not as administrator.
+            // the keystore ends up being owned by the Administrators group, so we manually set it to be owned by the vagrant user here.
+            // from the server's perspective the permissions aren't really different, this is just to reflect what we'd expect in the tests.
+            // when we run these commands as a role user we won't have to do this
+            sh.powershell(
+                installation.bin("elasticsearch-keystore.bat") + " create; " +
+                "$account = New-Object System.Security.Principal.NTAccount 'vagrant'; " +
+                "$acl = Get-Acl '" + installation.config("elasticsearch.keystore") + "'; " +
+                "$acl.SetOwner($account); " +
+                "Set-Acl '" + installation.config("elasticsearch.keystore") + "' $acl"
+            );
+        } else {
+            sh.bash("sudo -u " + ARCHIVE_OWNER + " " + installation.bin("elasticsearch-keystore") + " create");
+        }
+
+        assertThat(installation.config("elasticsearch.keystore"), file(File, ARCHIVE_OWNER, ARCHIVE_OWNER, p660));
+
+        final Result r = Platforms.WINDOWS
+            ? sh.powershell(installation.bin("elasticsearch-keystore.bat") + " list")
+            : sh.bash("sudo -u " + ARCHIVE_OWNER + " " + installation.bin("elasticsearch-keystore") + " list");
+        assertThat(r.stdout, containsString("keystore.seed"));
+
+        // cleanup for next test
+        rm(installation.config("elasticsearch.keystore"));
+    }
+
+    @Test
+    public void test50StartAndStop() {
+        assumeThat(installation, is(notNullValue()));
+
+        Archives.runElasticsearch(installation);
+
+        final String gcLogName = Platforms.LINUX
+            ? "gc.log.0.current"
+            : "gc.log";
+        assertTrue("gc logs exist", Files.exists(installation.logs.resolve(gcLogName)));
+        ServerUtils.runElasticsearchTests();
+
+        Archives.stopElasticsearch(installation);
+    }
+
+    @Test
+    public void test60AutoCreateKeystore() {
+        assumeThat(installation, is(notNullValue()));
+
+        assertThat(installation.config("elasticsearch.keystore"), file(File, ARCHIVE_OWNER, ARCHIVE_OWNER, p660));
+
+        final Shell sh = new Shell();
+        final Result result;
+        if (Platforms.WINDOWS) {
+            result = sh.powershell(installation.bin("elasticsearch-keystore.bat") + " list");
```

Comment:
I'm wondering if it would make sense to have an abstraction on top of `Shell` to call bin scripts in a platform independent way. I think it would make tests more readable e.x. having something like `ESShell().kestore().list()` to achieve this. Or maybe `EsShell().escalatePrivileges().keystore().list()` to make it run with administrator/root

Reply:
Agree, though I'd lean towards just doing it for the executable paths